### PR TITLE
kvserver: remove legacy snapshot and diff code

### PR DESCRIPTION
**Pull request by pav-kv**
This commit removes the proto and code for `RaftDataSnapshot`. This type was used for reporting replica snapshots and computing diffs, but now this functionality has been removed in favor of storage checkpoints and offline tooling.

Part of #21128
Epic: none

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/replica_consistency.go [hunk 1]
```diff
@@ -482,18 +480,18 @@ type replicaHash struct {
...
-	var r *Replica
-	var snap roachpb.RaftSnapshotData
-	lim := quotapool.NewRateLimiter("test", 1<<20, 1<<20)
-	if _, err := r.sha512(ctx, rd, store, &snap, roachpb.ChecksumMode_CHECK_FULL, lim); err != nil {
-		return roachpb.RaftSnapshotData{}, err
-	}
-	return snap, nil
+// ChecksumRangeForTesting returns a checksum over the KV data of the given
+// range. Only for testing.
+func ChecksumRangeForTesting(
```
Comment (pavelkalinnikov) [inline pkg/kv/kvserver/replica_consistency.go:485]: Move this to `_test.go` file, as @tbg suggested previously.

**Comment by erikgrinaker** [comment]
Let's hold off until we know what's up with 22.1 compatibility. I think we can still get rid of the diffs, because 22.1 will degrade gracefully if we omit them from the responses, but let's confirm.

File: pkg/kv/kvserver/helpers_test.go [hunk 2]
```diff
@@ -550,3 +550,16 @@ func WatchForDisappearingReplicas(t testing.TB, store *Store) {
 		}
 	}
 }
+
+// ChecksumRange returns a checksum over the KV data of the given range.
+func ChecksumRange(
+	ctx context.Context, desc roachpb.RangeDescriptor, snap storage.Reader,
+) ([]byte, error) {
+	var r *Replica // TODO(pavelkalinnikov): make this less ugly.
```
Comment (erikgrinaker) [inline pkg/kv/kvserver/helpers_test.go:558]: `sha512()` doesn't use the `Replica` descriptor for anything, so we can just rename and export it instead.

**Review by erikgrinaker — APPROVED** [review]
Good riddance!

As you may know, version-skipping upgrades were pushed back, so we don't need 22.1 compatibility for 23.1. In other words, let's merge this.

File: pkg/kv/kvserver/helpers_test.go (hunk 2 above)
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/helpers_test.go]: I'll see tomorrow if I can make this even cleaner, and will send a follow-up PR if so.

**Comment by pavelkalinnikov** [comment]
bors r=erikgrinaker

**Pull request merged**